### PR TITLE
gh-99947: Ensure unreported error is chained for SystemError during import

### PR DESCRIPTION
**Pull request by seberg**
Before this change the triggering exception was dropped.

* Issue: gh-99947

### CODE REVIEW

File: Lib/test/test_importlib/extension/test_loader.py [hunk 1]
```diff
@@ -351,9 +351,13 @@ def test_bad_modules(self):
                 ]:
             with self.subTest(name_base):
                 name = self.name + '_' + name_base
-                with self.assertRaises(SystemError):
+                with self.assertRaises(SystemError) as cm:
                     self.load_module_by_name(name)
 
+                # If there is an unreported exception, it should be chained
```
Comment (brettcannon) [inline Lib/test/test_importlib/extension/test_loader.py:357]:
```suggestion
                # If there is an unreported exception, it should be chained with the `SystemError`.
```

File: Lib/test/test_importlib/extension/test_loader.py [hunk 2]
```diff
@@ -351,9 +351,13 @@ def test_bad_modules(self):
...
+                if "unreported_exception" in name_base:
+                    self.assertIsNot(cm.exception.__cause__, None)
```
Comment (brettcannon) [inline Lib/test/test_importlib/extension/test_loader.py]:
```suggestion
                    self.assertIsNotNone(cm.exception.__cause__)
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
Some minor tweaks, but otherwise I'm good with the proposed change.

File: Misc/NEWS.d/next/C API/2022-12-02-09-31-19.gh-issue-99947.Ski7OC.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+SystemErrors on import now chain the original unexpected exception
```
Comment (brettcannon) [inline Misc/NEWS.d/next/C API/2022-12-02-09-31-19.gh-issue-99947.Ski7OC.rst]:
```suggestion
Raising SystemError on import will now have its cause be set to the original unexpected exception.
```

**Review by ericsnowcurrently — COMMENTED** [review]
I agree with Brett. :)

**Comment by seberg** [comment]
Adopted the improvements, thanks.  Its a minor help with debugging, but would have saved me time twice in the past 1-2 months.

**Comment by seberg** [comment]
"I have made the requested changes; please review again"

File: Lib/test/test_importlib/extension/test_loader.py [hunk 4]
```diff
@@ -351,9 +351,14 @@ def test_bad_modules(self):
                 ]:
             with self.subTest(name_base):
                 name = self.name + '_' + name_base
-                with self.assertRaises(SystemError):
+                with self.assertRaises(SystemError) as cm:
                     self.load_module_by_name(name)
 
+                # If there is an unreported exception, it should be chained
+                # with the `SystemError`
```
Comment (brettcannon) [inline Lib/test/test_importlib/extension/test_loader.py]:
```suggestion
                # with the `SystemError`.
```

**Review by brettcannon — APPROVED** [review]

**Pull request merged**